### PR TITLE
Display percentage of correct submissions in table.

### PR DESCRIPTION
**Pull request by dsjen**
screen shot 2015-03-16 at 6 04 42 pm

@jab5569 @brianhw @mulby 

FYI: @shnayder @lamagnifica @andywaldrop

### CODE REVIEW

**Comment by lamagnifica** [comment]
:+1: with an observation: the space between the Total and Percentage Correct columns seems much greater than the space between the Correct, Incorrect, and Total columns. I don't know if that is something that can be changed, but it draws my eye. @andywaldrop thoughts?

**Comment by andywaldrop** [comment]
:+1:  Agree with @lamagnifica if we can make the columns more consistently spaced.

**Comment by dsjen** [comment]
I tried fiddling with the column width, but it's quite difficult to get right--sorry!

File: acceptance_tests/mixins.py [hunk 1]
```diff
@@ -318,8 +318,11 @@ def fulfill_loading_promise(self, css_selector):
         ).fulfill()
 
     def build_display_percentage(self, count, total):
-        percent = count / total * 100
-        return '{:.1f}%'.format(percent) if percent >= 1.0 else '< 1%'
+        if total:
```
Comment (johnalbaker) [inline acceptance_tests/mixins.py]: What if 'count' doesn't exist? Eg, aren't both 'count' and 'total' needed?

File: acceptance_tests/mixins.py [hunk 2]
```diff
@@ -318,8 +318,11 @@ def fulfill_loading_promise(self, css_selector):
         ).fulfill()
 
     def build_display_percentage(self, count, total):
```
Comment (johnalbaker) [inline acceptance_tests/mixins.py:320]: Off the top of your head, are we ensuring that 'totat' and 'count' are integers elsewhere in the code?

**Comment by johnalbaker** [comment]
Other than a few questions, [image: :+1:] LGTM.

On Mon, Mar 16, 2015 at 6:06 PM, Dennis Jen notifications@github.com
wrote:

> [image: screen shot 2015-03-16 at 6 04 42 pm]
> https://cloud.githubusercontent.com/assets/5265058/6677493/f0030cce-cc06-11e4-88ca-b024ec206cb1.png
> 
> @jab5569 https://github.com/jab5569 @brianhw
> https://github.com/brianhw @mulby https://github.com/mulby
> 
> FYI: @shnayder https://github.com/shnayder @lamagnifica
> https://github.com/lamagnifica @andywaldrop
> 
> ## https://github.com/andywaldrop
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/edx/edx-analytics-dashboard/pull/291
> Commit Summary
> - Display percentage of correct submissions in table.
> 
> File Changes
> - _M_ acceptance_tests/mixins.py
>   https://github.com/edx/edx-analytics-dashboard/pull/291/files#diff-0
>   (7)
> - _M_ acceptance_tests/test_course_performance.py
>   https://github.com/edx/edx-analytics-dashboard/pull/291/files#diff-1
>   (33)
> - _M_
>   analytics_dashboard/static/js/performance-graded-content-assignment-main.js
>   https://github.com/edx/edx-analytics-dashboard/pull/291/files#diff-2
>   (7)
> - _M_
>   analytics_dashboard/static/js/performance-graded-content-assignment-types-main.js
>   https://github.com/edx/edx-analytics-dashboard/pull/291/files#diff-3
>   (7)
> - _M_ analytics_dashboard/static/js/test/specs/data-table-view-spec.js
>   https://github.com/edx/edx-analytics-dashboard/pull/291/files#diff-4
>   (6)
> - _M_ analytics_dashboard/static/js/views/data-table-view.js
>   https://github.com/edx/edx-analytics-dashboard/pull/291/files#diff-5
>   (7)
> 
> Patch Links:
> - https://github.com/edx/edx-analytics-dashboard/pull/291.patch
> - https://github.com/edx/edx-analytics-dashboard/pull/291.diff
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/edx/edx-analytics-dashboard/pull/291.

## 

John Baker
edX | Analytics Architect
141 Portland Street, 8th Floor
Cambridge, MA 02139

john@edx.org mcoyne@edx.org
http://www.edx.org

File: acceptance_tests/mixins.py (hunk 2 above)
Comment (dsjen) [inline acceptance_tests/mixins.py:320]: The data comes from the API, and I don't know that we validate or cast it when it comes in.

File: acceptance_tests/mixins.py (hunk 1 above)
Comment (dsjen) [inline acceptance_tests/mixins.py]: It wouldn't hurt to check `count`, and I'll add in a check for that.  Thanks!

**Pull request merged**